lib/models/ObjectMDLocation.js

```diff
@@ -33,9 +33,25 @@ class ObjectMDLocation {
         return this._data.dataStoreName;
     }
 
+    /**
+     * Update data location with new info
+     *
+     * @param {object} location - single data location info
+     * @param {string} location.key - data backend key
+     * @param {string} location.dataStoreName - type of data store
+     * @param {number} [location.cryptoScheme] - if location data is
+     * encrypted: the encryption scheme version
+     * @param {string} [location.cipheredDataKey] - if location data
+     * is encrypted: the base64-encoded ciphered data key
+     * @return {ObjectMDLocation} return this
+     */
     setDataLocation(location) {
         this._data.key = location.key;
         this._data.dataStoreName = location.dataStoreName;
+        if (location.cryptoScheme) {
+            this._data.cryptoScheme = location.cryptoScheme;
+            this._data.cipheredDataKey = location.cipheredDataKey;
+        }
```

Comment:
Could you do this all with Object.assign?
```suggestion
        this._data = Object.assign(this._data, location);
```